Bugfix/zenko 4912

benzekrimaha opened the pull request:
Issue : ZENKO-4912

bert-e commented:
# Hello benzekrimaha,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4912 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.10.2`


Please check the `Fix Version/s` of ZENKO-4912, or the target
branch of this pull request.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand commented on monitoring/mongodb/alerts.test.yaml:
```diff
@@ -333,10 +333,6 @@ tests:
-      - series: mongodb_rs_members_state{namespace="zenko", rs_nm="rs0", pod="data-db-mongodb-sharded-mongos-0", member_state="SECONDARY"}
-        values: 0x10
-      - series: mongodb_rs_members_state{namespace="zenko", rs_nm="rs0", pod="data-db-mongodb-sharded-mongos-1", member_state="SECONDARY"}
-        values: 0x10
```
why remove these?

to test the behavior, we need input series which are consistent with the actual production situation : so need to have multiple series, including all the labels you use to make sure we validate it works in exactly the right cases.

→ we aggregate by `rs_nm`, so we should check this aggregation works
→ this typically happens in 3-node cluster, so we should have the 3 series for mongo0, mongo1 and mong2

in this specific case, the fix you introduce is to add an `instance` : so the series should include it as well, possible with extra series to validate we do not unexpectedly "merge" what should be separate

benzekrimaha replied:
Done here : https://github.com/scality/Zenko/pull/2160/commits/5bbb61af3a8c86e18d2f5018a175fe9ac03b315e#diff-94c90083eba829343ee5745d0056d887a02d888847ea33af739a3ef5c3ae92c6R336-R345

francoisferrand commented on monitoring/mongodb/alerts.yaml:
```diff
@@ -183,7 +183,7 @@ groups:
   - alert: MongoDbRSNotSynced
     expr: |
-      sum by (rs_nm) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) != (${replicas} - 1)
+      count by (rs_nm, instance) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) != (${replicas} - 1)
```
what is `instance` ?

benzekrimaha replied:
the instance on the metrics is typically the `IP:PORT => instance="10.233.226.186:9216"`, it is being used as each pod will be reporting the same metric with the same value ( the state which is 2 ) resulting in a 3 node for example in a returned value of 12 without the additionnal filtering

francoisferrand replied:
if we count by instance, we will only get 1 series for each instance : so the count will mostly by `1`...

benzekrimaha replied:
No , when we count by the instance we indeed have the right number cf ( this was done on a 6 nodes ) : 
![image](https://github.com/user-attachments/assets/043f2b93-f635-4c7e-af2b-9ec55c86db66)

benzekrimaha replied:
No the count by the instance and rs_nm returns indeed the right number of secondary members cf : ( this was done on a 6 nodes artesca ) 
<img width="1312" alt="image" src="https://github.com/user-attachments/assets/22d2063d-2e3e-4100-96a8-ec02367827ff">

francoisferrand replied:
but why does it work? AFAIK, each pod should be a single instance, something fishy here...
(and the exemple in the test does not really match the graph you show here)

can you share the actual metrics: e.g. in explore, check which metrics series we get with

```
mongodb_rs_members_state{namespace="zenko", pod=~"data-db-mongodb_sharded-shard0-data"}
```

benzekrimaha replied:
with : 
`mongodb_rs_members_state{namespace="zenko", pod=~"data-db-mongodb-sharded-shard0-data.*"}`\

this is what is returned ( for example in fact we get 3 metrics 2 are secondary members and 1 for the primary ) : 
`{__name__="mongodb_rs_members_state", cl_id="65d9fdecfdcca86bb7cd2b98", container="metrics", endpoint="metrics", instance="10.233.178.136:9216", job="zenko/data-db-mongodb-sharded-shard0-data", member_idx="data-db-mongodb-sharded-shard0-data-1.data-db-mongodb-sharded-headless.zenko.svc.cluster.local:27017", member_state="SECONDARY", namespace="zenko", pod="data-db-mongodb-sharded-shard0-data-0", prometheus="metalk8s-monitoring/prometheus-operator-prometheus", rs_nm="data-db-mongodb-sharded-shard-0", rs_state="1"}`


and the value returned is 2 ( the state as it's a secondary ) => 
![image](https://github.com/user-attachments/assets/745c5e70-574f-4792-826a-57177861ef50)

francoisferrand replied:
as we can see in the "full" metrics, each pod reports the status (primary/secondary/...) of every member of the cluster participating in the replicaset, as identified by the `member_idx` label

francoisferrand commented on monitoring/mongodb/alerts.test.yaml:
```diff
@@ -333,12 +333,16 @@ tests:
+        values: 2x10
+      - series: mongodb_rs_members_state{namespace="zenko", rs_nm="rs0", pod="data-db-mongodb-sharded-mongos-2", member_state="SECONDARY", instance="10.233.226.139:9216"}
+        values: 2x10
+      - series: mongodb_rs_members_state{namespace="zenko", rs_nm="rs1", pod="data-db-mongodb-sharded-mongos-0", member_state="SECONDARY", instance="10.233.226.140:9216"}
+        values: 2x10
+      - series: mongodb_rs_members_state{namespace="zenko", rs_nm="rs1", pod="data-db-mongodb-sharded-mongos-1", member_state="SECONDARY", instance="10.233.226.140:9216"}
```
`instance` is the IP of the pod which exposes the metric: so multiple `pod` should not have the same value for `instance`, right?

benzekrimaha replied:
yes , this is why we have an example with multiple IP and one with the same one ( the first fires alerts while the second doesn't )

francoisferrand replied:
it should not just be an "example" but also match reality : we should have metrics which are similar to what is in a real cluster, and then introduce changes to simulate various error conditions...

francoisferrand commented on monitoring/mongodb/alerts.test.yaml:
```diff
@@ -333,23 +333,30 @@ tests:
+        values: 2x10
+      - series: mongodb_rs_members_state{cl_id="65d9fdecfdcca86bb7cd2b98", container="metrics", endpoint="metrics", instance="10.233.178.160:9216", job="zenko/data-db-mongodb-sharded-shard0-data", member_idx="data-db-mongodb-sharded-shard0-data-1.data-db-mongodb-sharded-headless.zenko.svc.cluster.local:27017", member_state="SECONDARY", namespace="zenko", pod="data-db-mongodb-sharded-shard0-data-0 ...
+        values: 2x10
+      - series: mongodb_rs_members_state{cl_id="65d9fdecfdcca86bb7cd2b98", container="metrics", endpoint="metrics", instance="10.233.178.160:9216", job="zenko/data-db-mongodb-sharded-shard0-data", member_idx="data-db-mongodb-sharded-shard0-data-2.data-db-mongodb-sharded-headless.zenko.svc.cluster.local:27017", member_state="SECONDARY", namespace="zenko", pod="data-db-mongodb-sharded-shard0-data-0 ...
```
for clarity, remove the labels we do not use (and basically ignore): container, endpoint, cl_id, instance, prometheus, job...
 → focus on the relevant labels

benzekrimaha commented on monitoring/mongodb/alerts.yaml:
```diff
@@ -183,10 +183,10 @@ groups:
   - alert: MongoDbRSNotSynced
     expr: |
-      sum by (rs_nm) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) != (${replicas} - 1)
+      count by (rs_nm, pod) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY", rs_state="1"}) != (${replicas} - 1)
```
The pod was added , as when we don't add this additional filtering we end up with a total of 3 instead of 2 , this is explained by the fact that for each pod we have 2 metrics exported , given the fact that these metrics actually have the same pos, namespace, rs_state and state member it is concidered as 1 metric so gives 1 x 3 which results in 3 , when filtering by the pod as well we make sure that for each pod the exported metrics return indeed 2 secondaries. please don't hesitate to challenge this

francoisferrand replied:
* why do we filter on `rs_state=1` : are the wrong states not covered by another alert, or is this covering extra case?
* if there is an issue, this will however trigger multiple alerts: typically, if a single (mongo) pod is down, each of the remaining pod will generate an alert, and 2 alerts will be raised instead of 1 → should avoid this duplication
* the `pod` is actually not relevant -and possibly misleading- as a parameter of the alert: as it seems to indicate an issue with this pod....when actually the issue is with the "remaining" pod: the one(s) which may not actually trigger the alert... → need to make sure the `pod` label does not end up in the final time serie

maybe something like this would solve the issue:

```suggestion
      min by(rs_nm)(
         count by (rs_nm, pod) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) != (${replicas} - 1)
      )
```

benzekrimaha commented on monitoring/mongodb/alerts.test.yaml:
```diff
@@ -333,23 +333,31 @@ tests:
-      - series: mongodb_rs_members_state{namespace="zenko", rs_nm="rs0", pod="data-db-mongodb-sharded-mongos-1", member_state="SECONDARY"}
-        values: 0x10
-      - series: mongodb_rs_members_state{namespace="zenko", rs_nm="rs0", pod="data-db-mongodb-sharded-mongos-2", member_state="SECONDARY"}
+      - series: mongodb_rs_members_state{namespace="zenko", pod="data-db-mongodb-sharded-shard0-data-0", member_state="PRIMARY", rs_nm="data-db-mongodb-sharded-shard-0", rs_state="1",  member_idx="data-db-mongodb-sharded-shard0-data-0.data-db-mongodb-sharded-headless.zenko.svc.cluster.local:27017"}
```
member_idx has been added as it influences the calculous , without it we get a result of 1 , but once it is provided ( matching the real life metrics within the plateforms ) the calculous becomes 2 matching the right number of secondaries in the replicaset / pod

francoisferrand replied:
it does not _influence the computation_ : only if you don't include it, you actually provide the exact same time-series multiple times and you end up actually having a single serie.... hence the result of the computation.

francoisferrand commented on VERSION:
```diff
@@ -1,4 +1,4 @@
-VERSION="2.10.1"
+VERSION="2.10.2"
```
reminder to rebase: this should be the last commit of the PR (where we will make the release)

benzekrimaha commented on monitoring/mongodb/alerts.yaml:
```diff
@@ -183,10 +183,10 @@ groups:
   - alert: MongoDbRSNotSynced
     expr: |
-      sum by (rs_nm) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) != (${replicas} - 1)
+      floor(avg by(rs_nm)(count by (rs_nm, pod)(mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}))) != (${replicas} - 1) 
```
went for the floor avg as we want to know when the min or max is different from the expected secondaries number

francoisferrand replied:
to ensure we don't _miss_ any alert, we should do the boolean condition before the final aggregation.
Otherwise, we may be unlucky and different pod metrics may average with the right value....

```
<aggregation operator> by(rs_nm) (
    count by(rs_nm, pod) (...) != ${replicas} - 1
)
```

→ if no `pod` triggers the alert, the aggregation will have nothing to aggregate, and no series will trigger the alert
→ if one (or more) `pod` triggers the alert, then the aggregation operation is used to filter-out the `pod` label, and (possibly, but not mandatory) generate an aggregate `value` for the alert (`min` / `avg` / `max` ... number of replicas : though there could always be corner cases where the resulting value is unexecpted... ; if we don't want a value, just using `group` should work)

francoisferrand commented on monitoring/mongodb/alerts.yaml:
```diff
@@ -183,10 +183,10 @@ groups:
   - alert: MongoDbRSNotSynced
     expr: |
-      sum by (rs_nm) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) != (${replicas} - 1)
+      group by(rs_nm)(count by (rs_nm, pod)(mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) != (${replicas} - 1) )
```
nit: indent to improve readabiltiy...

```suggestion
      group by(rs_nm) (
        count by(rs_nm, pod) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"})
        != (${replicas} - 1)
      )
```

francoisferrand commented on monitoring/mongodb/alerts.yaml:
```diff
@@ -183,10 +183,10 @@ groups:
       severity: warning
     annotations:
-      description: "MongoDB replica set `{{ $labels.rs_nm }}` is not in the expected state. It currently has `{{ $value }}` SECONDARY members instead of the expected number. Please ensure that all instance are running properly."
+      description: "MongoDB replica set `{{ $labels.rs_nm }}` is not in the expected state. Please ensure that all instances are running properly."
```
nit: description should still report on "what" is wrong (i.e. number of SECONDARY)

```suggestion
      description: "MongoDB replica set `{{ $labels.rs_nm }}` is not in the expected state. It does not have the expected number of SECONDARY members. Please ensure that all instances are running properly."
```

francoisferrand commented on monitoring/mongodb/alerts.test.yaml:
```diff
@@ -333,23 +333,32 @@ tests:
+      - series: mongodb_rs_members_state{namespace="zenko", pod="data-db-mongodb-sharded-shard0-data-2", member_state="SECONDARY", rs_nm="data-db-mongodb-sharded-shard-0", member_idx="data-db-mongodb-sharded-shard0-data-1.data-db-mongodb-sharded-headless.zenko.svc.cluster.local:27017"}
+        values: 2x10
+      - series: mongodb_rs_members_state{namespace="zenko", pod="data-db-mongodb-sharded-shard0-data-2", member_state="(not reachable/healthy)", rs_nm="data-db-mongodb-sharded-shard-0", member_idx="data-db-mongodb-sharded-shard0-data-2.data-db-mongodb-sharded-headless.zenko.svc.cluster.local:27017"}
+        values: 2 _ _ _ _ _ _ _ _ _
```
when this serie disappears, the other series will actually change as well: as the remaining pods will not report on its status anymore... and the state reporte looks strange :/
* On "startup", we should have series : everything is running fine (3 pods having the status of 3 members)
* Then one pod "crashes" : the 3 associated series stop
* Then the other pods update their "vision" of the cluster : they either report a different member state (not reachable?) or drop the member completely (not sure exactly what happens)

benzekrimaha replied:
from what I saw the state is (not reachable/healthy) , the test was updated accordingly

francoisferrand replied:
every metric from `pod="data-db-mongodb-sharded-shard0-data-2"` should be removed when we simulate the pod crashing...

so i guess time series should look something like this:

```
# First pod
- series: member_state{pod="data-0", state="PRIMARY", member_idx="data-0"}
   values: 1x10 1x10
- series: member_state{pod="data-0", state="SECONDARY", member_idx="data-1"}
   values: 2x10 2x10
- series: member_state{pod="data-0", state="SECONDARY", member_idx="data-2"}
   values: 2x10 stale      # `stale` (i.e. not present) after the 10s sample
- series: member_state{pod="data-0", state="(not reachable/healthy)", member_idx="data-2"}
   values: _x10 8x10      # appears from the 11s sample
# Second pod
- series: member_state{pod="data-1", state="PRIMARY", member_idx="data-0"}
   values: 1x10 1x10
- series: member_state{pod="data-1", state="SECONDARY", member_idx="data-1"}
   values: 2x10 2x10
- series: member_state{pod="data-1", state="SECONDARY", member_idx="data-2"}
   values: 2x10 stale      # `stale` (i.e. not present) after the 10s sample
- series: member_state{pod="data-1", state="(not reachable/healthy)", member_idx="data-2"}
   values: _x10 8x10      # appears from the 11s sample
# Third pod : stops responding after 10th sample
- series: member_state{pod="data-2", state="PRIMARY", member_idx="data-0"}
   values: 1x10 stale
- series: member_state{pod="data-2", state="SECONDARY", member_idx="data-1"}
   values: 2x10 stale
- series: member_state{pod="data-2", state="SECONDARY", member_idx="data-2"}
   values: 2x10 stale
```

benzekrimaha replied:
Providing this :  series: member_state{pod="data-0", state="SECONDARY", member_idx="data-2"}
   values: 2x10 stale      # `stale` (i.e. not present) after the 10s sample
   
  will cause the expression to return 2 => expected number of secondaries as it's not taking the state="(not reachable/healthy)" metric

francoisferrand replied:
AFAIU, `stale` removes the metric series, which should be what really happens:
* the serie with state label equal to `secondary` (and value `2`) stops
* another serie starts with state label equal to `(not reachable/healthy)` and value `8`

(since one label value changes, this is really another serie : even if [as human] we think it is the same one...)

francoisferrand commented on monitoring/mongodb/alerts.test.yaml:
```diff
@@ -333,23 +333,32 @@ tests:
+      - series: mongodb_rs_members_state{namespace="zenko", pod="data-db-mongodb-sharded-shard0-data-0", member_state="SECONDARY", rs_nm="data-db-mongodb-sharded-shard-0", member_idx="data-db-mongodb-sharded-shard0-data-1.data-db-mongodb-sharded-headless.zenko.svc.cluster.local:27017"}
+        values: 2x10
+      - series: mongodb_rs_members_state{namespace="zenko", pod="data-db-mongodb-sharded-shard0-data-0", member_state="(not reachable/healthy)", rs_nm="data-db-mongodb-sharded-shard-0", member_idx="data-db-mongodb-sharded-shard0-data-2.data-db-mongodb-sharded-headless.zenko.svc.cluster.local:27017"}
+        values: 2 _ _ _ _ _ _ _ _ _
```
here the member "2" is initially not avaialble, then the series stops : is this really what you meant?

benzekrimaha replied:
I'm trying to keep the same values as in the `pod="data-db-mongodb-sharded-shard0-data-2"`, as it's the one with the secondary down

francoisferrand reviewed (commented):
LGTM, but alert test fail...

francoisferrand commented on monitoring/mongodb/alerts.test.yaml:
```diff
@@ -329,27 +329,39 @@ tests:
+      - series: mongodb_rs_members_state{namespace="zenko", pod="data-db-mongodb-sharded-shard0-data-1", member_state="(not reachable/healthy)", rs_nm="data-db-mongodb-sharded-shard-0", member_idx="shard0-data-2"}
+        values: stale _x8 8x10
+      - series: mongodb_rs_members_state{namespace="zenko", pod="data-db-mongodb-sharded-shard0-data-2", member_state="PRIMARY", rs_nm="data-db-mongodb-sharded-shard-0", member_idx="shard0-data-0"}
+        values: 1x10 stale
+      - series: mongodb_rs_members_state{namespace="zenko", pod="data-db-mongodb-sharded-shard0-data-2", member_state="SECONDARY", rs_nm="data-db-mongodb-sharded-shard-0", member_idx="shard0-data-1"}
+        values: 2x10 stale
```
should be `1x8 stale` and `2x8 stale` for consistency with the third metrics serie (`2x8 stale`)

francoisferrand commented on monitoring/mongodb/alerts.test.yaml:
```diff
@@ -329,27 +329,39 @@ tests:
     alert_rule_test:
       - alertname: MongoDbRSNotSynced
-        eval_time: 10m
+        eval_time: 19m
```
should keep a check at different times before (5m → no alert, 10m → no alert, 18m → no alert)

francoisferrand reviewed (approved):
LGTM, minor test improvements : please rebase/cleanup before approving

benzekrimaha committed:
fixup on MongoDbRSNotSynced expression

MongoDbRSNotSynced firing when it shouldn't because today we sum up the members state ,
as we are calculating the number of secondaries, we end up with a higher value |
than the expected one to have the right value an additional filtering based
on the instance have been introduced as well
Issue: ZENKO-4912

benzekrimaha committed:
fixup on MongoDbRSNotSynced test

Issue: ZENKO-4912

benzekrimaha committed:
Bump zenko to 2.10.2

Issue: ZENKO-4912

benzekrimaha commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/Zenko/commit/80ff4c7c93bda3e3b070eeea4e88c18103bd80f3) did not succeed in branch bugfix/ZENKO-4912.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.10`



The following branches will **NOT** be impacted:

* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# Queue build failed




The corresponding build for the queue failed:

- Checkout the [status page](https://bert-e.scality.net/scality/zenko).
- Identify the failing build and review the logs.
- If no issue is found, re-run the build.
- If an issue is identified, checkout the steps below to remove
  the pull request from the queue for further analysis and maybe rebase/merge.

<details>
  <summary><b>Remove the pull request from the queue</b></summary>

- Add a `/wait` comment on this pull request.
- Click on login on the [status page](https://bert-e.scality.net/scality/zenko).
- Go into the [manage](https://bert-e.scality.net/scality/zenko/manage) page.
- Find the option called `Rebuild the queue` and click on it.
  Bert-E will loop again on all pull requests to put the valid ones
  in the queue again, while skipping the one with the `/wait` comment.
- Wait for the new queue to merge, then merge/rebase your pull request
  with the latest changes to then work on a proper fix.
- Once the issue is fixed, delete the `/wait` comment and
  follow the usual process to merge the pull request.

</details>

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.10`



The following branches have **NOT** changed:

* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue ZENKO-4912.

Goodbye benzekrimaha.
